[pkg/stanza/fileconsumer] TestExcludeOlderThanFilter/file_not_present test failing on Windows

### Component(s)

pkg/stanza

### Describe the issue you're reporting

[Failed CI/CD run](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8711264614/job/23894973542?pr=31915)

Failure output:
```
=== Failed
=== FAIL: fileconsumer/matcher/internal/filter TestExcludeOlderThanFilter/file_not_present (0.00s)
    exclude_test.go:95: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/internal/filter/exclude_test.go:95
        	Error:      	Error "CreateFile C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestExcludeOlderThanFilterfile_not_present318488423\\001\\b.log: The system cannot find the file specified." does not contain "b.log: no such file or directory"
        	Test:       	TestExcludeOlderThanFilter/file_not_present

=== FAIL: fileconsumer/matcher/internal/filter TestExcludeOlderThanFilter (0.02s)

=== FAIL: fileconsumer/matcher/internal/filter TestExcludeOlderThanFilter/file_not_present (re-run 1) (0.00s)
    exclude_test.go:95: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/matcher/internal/filter/exclude_test.go:95
        	Error:      	Error "CreateFile C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestExcludeOlderThanFilterfile_not_present764582634\\001\\b.log: The system cannot find the file specified." does not contain "b.log: no such file or directory"
        	Test:       	TestExcludeOlderThanFilter/file_not_present

=== FAIL: fileconsumer/matcher/internal/filter TestExcludeOlderThanFilter (re-run 1) (0.01s)

DONE 2 runs, 3282 tests, 13 skipped, 4 failures in 104.549s
make[2]: *** [../../Makefile.Common:126: test] Error 1
make[1]: *** [Makefile:165: pkg/stanza] Error 2
make[1]: *** Waiting for unfinished jobs....
✓  . (1.353s)

DONE 121 tests in 5.801s
make[1]: Leaving directory 'D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
make: *** [Makefile:113: gotest] Error 2
```

Potentially related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32299, where the solution was to use `filepath.Join` for paths to work properly on Windows.

--- a/pkg/stanza/fileconsumer/matcher/internal/filter/exclude_test.go
+++ b/pkg/stanza/fileconsumer/matcher/internal/filter/exclude_test.go
@@ -6,6 +6,7 @@ package filter
 import (
 	"os"
 	"path/filepath"
+	"runtime"
 	"testing"
 	"time"
 
@@ -21,8 +22,9 @@ func TestExcludeOlderThanFilter(t *testing.T) {
 		fileMTimes       []time.Time
 		excludeOlderThan time.Duration
 
-		expect      []string
-		expectedErr string
+		expect             []string
+		expectedErr        string
+		expectedWindowsErr string
 	}{
 		"no_files": {
 			files:            []string{},
@@ -61,8 +63,9 @@ func TestExcludeOlderThanFilter(t *testing.T) {
 			fileMTimes:       []time.Time{twoHoursAgo, {}},
 			excludeOlderThan: 3 * time.Hour,
 
-			expect:      []string{"a.log"},
-			expectedErr: "b.log: no such file or directory",
+			expect:             []string{"a.log"},
+			expectedErr:        "b.log: no such file or directory",
+			expectedWindowsErr: "b.log: The system cannot find the file specified.",
 		},
 	}
 
@@ -92,7 +95,11 @@ func TestExcludeOlderThanFilter(t *testing.T) {
 			f := ExcludeOlderThan(tc.excludeOlderThan)
 			result, err := f.apply(items)
 			if tc.expectedErr != "" {
-				require.ErrorContains(t, err, tc.expectedErr)
+				if runtime.GOOS == "windows" {
+					require.ErrorContains(t, err, tc.expectedWindowsErr)
+				} else {
+					require.ErrorContains(t, err, tc.expectedErr)
+				}
 			} else {
 				require.NoError(t, err)
 			}

